app/scripts/modules/core/src/managed/Environments.tsx

```diff
@@ -99,46 +140,46 @@ export function Environments({ app }: IEnvironmentsProps) {
     );
   }
 
-  const totalContentWidth = isFiltersOpen ? CONTENT_WIDTH + 248 + 'px' : CONTENT_WIDTH + 'px';
-
   return (
-    <div style={{ width: '100%' }}>
-      <span>For there shall be no greater pursuit than that towards desired state.</span>
-      <div style={{ maxWidth: totalContentWidth, display: 'flex' }}>
-        {/* No filters for now but this is where they will go */}
-        <div className={styles.mainContent} style={{ flex: `0 1 ${totalContentWidth}` }}>
-          <div className={styles.artifactsColumn}>
-            <ColumnHeader text="Artifacts" icon="artifact" />
-            <ArtifactsList
-              artifacts={artifacts}
-              selectedVersion={selectedVersion}
-              versionSelected={clickedVersion => {
-                if (!isEqual(clickedVersion, selectedVersion)) {
-                  go(selectedVersion ? '.' : '.artifactVersion', clickedVersion);
-                }
-              }}
-            />
-          </div>
-          <div className={styles.environmentsColumn}>
-            {!selectedVersion && (
-              <>
+    <div className={styles.mainContent}>
+      <div className={styles.artifactsColumn}>
+        <ColumnHeader text="Artifacts" icon="artifact" />
+        <ArtifactsList
+          artifacts={artifacts}
+          selectedVersion={selectedVersion}
+          versionSelected={clickedVersion => {
+            if (!isEqual(clickedVersion, selectedVersion)) {
+              go(selectedVersion ? '.' : '.artifactVersion', clickedVersion);
+            }
+          }}
+        />
+      </div>
+      <div className={styles.environmentsColumn}>
+        {overviewPaneTransition.map(
+          ({ item: show, key, props }) =>
+            show && (
+              <animated.div key={key} className={styles.environmentsPane} style={props}>
                 <ColumnHeader text="Environments" icon="environment" />
                 <EnvironmentsList {...{ environments, artifacts, resourcesById }} />
-              </>
-            )}
-            {selectedVersion && (
-              <ArtifactDetail
-                application={app}
-                name={selectedVersion.name}
-                type={selectedVersion.type}
-                version={selectedVersionDetails}
-                allVersions={selectedArtifactDetails.versions}
-                resourcesByEnvironment={resourcesByEnvironment}
-                onRequestClose={() => go('^')}
-              />
-            )}
-          </div>
-        </div>
+              </animated.div>
+            ),
+        )}
+        {detailPaneTransition.map(
+          ({ item, key, props }) =>
+            item.selectedVersion && (
```

Comment:
passing _all_ the data we need for rendering `ArtifactDetail` is nasty, and though it works I don't like it. The reason we need it is because referencing the usual `selectedVersion` / etc. would mean that the moment you hit close on the detail pane `selectedVersion` would become `null` and `ArtifactDetail` would have no data to use for rendering during its leave transition. That's fine for experimental purposes but should change for longer term use.

If we decide this transition is useful and worth keeping, I'll be encapsulating both the scrollable two column layout _and_ all this detail pane transition code inside a `<ListDetailLayout />` component that will let `ArtifactDetail` render itself via a render prop. That will mean we can use _older_ render prop callbacks to retain access to the previously selected version data without needing to pass things around through transition hooks. psuedo-example:

```tsx
// Because selectedVersion etc. are held inside the render prop's closure,
// we can hold onto its reference and use it to re-render `ArtifactDetail`
// with old props during the leave transition.
<ListDetailLayout
  renderListPane={() => { /* ... */ }}
  renderDetailPane={() =>
    <ArtifactDetail
      application={app}
      name={selectedVersion.name}
      type={selectedVersion.type}
      version={selectedVersionDetails}
      allVersions={selectedArtifactDetails.versions}
      resourcesByEnvironment={resourcesByEnvironment}
      onRequestClose={() => go('^')}
    />
  }
>
  {/* render initial/overview pane as a child? */}
</ListDetailLayout>
```